packages/editor-ui/src/stores/history.ts

```diff
@@ -0,0 +1,105 @@
+import { AddConnectionCommand, COMMANDS, RemoveConnectionCommand } from './../models/history';
+import { BulkCommand, Command, Undoable, MoveNodeCommand } from "@/models/history";
+import { STORES } from "@/constants";
+import { HistoryState } from "@/Interface";
+import { defineStore } from "pinia";
+
+const STACK_LIMIT = 100;
+
+export const useHistoryStore = defineStore(STORES.HISTORY, {
+	state: (): HistoryState => ({
+		undoStack: [],
+		redoStack: [],
+		currentBulkAction: null,
+		bulkInProgress: false,
+	}),
+	getters: {
+		currentBulkContainsConnectionCommand() {
+			return (command: AddConnectionCommand): boolean => {
```

Comment:
not sure what the exact problem here you are solving is.. but maybe this could be more generic.. if each class had a `compareTo` method like in Java or a `toString` representation, you can use that to check if an instance of that event exists in history